### PR TITLE
[travis] Test with local components instead of waiting for the subtree-splitter when possible

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

Should be ready - a first step that allows testing PR with the proposed patch. Works only when deps are resolved to the same branch. But this opens the way for further improvements.

### CODE REVIEW

File: .travis.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+branch=$1
+if [ "x" = "x$branch" ]; then
```
Comment (xabbuh) [inline .travis.sh]: What about using `if [ -z "$branch" ]` instead?

File: .travis.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+fi
+shift
+components="$*"
+if [ "x" = "x$components" ]; then
+    echo 'Usage: branch dir1 dir2 ... dirN'
+    exit 1
+fi
+echo '{"packages": {' > packages.json
+for c in $components; do
+    sed -i -z 's#^{\(\s*\)\("name"\)#{\1"repositories": \[{ "type": "composer", "url": "file://'$(pwd)'/" }\],\1\2#' $c/composer.json
```
Comment (stof) [inline .travis.sh]: looks like you are adding the same composer repository 30 times here approx (once per subtree repo). This seems useless. It should be added only once outside the loop

**Comment by stof** [comment]
Your script does not work. See the Travis build

**Comment by nicolas-grekas** [comment]
Tests are green now

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**